### ISSUE TITLE
Scheduled tasks may not run on clustered environments

### ISSUE THREAD

**Issue by gbbirkisson** [enonic/xp#9048]
One customer cannot get scheduled tasks to work in a clustered environment. Their task does not run. Looking at the logs we see this:

```
Audit log failed
java.lang.NullPointerException: null
	at com.enonic.xp.impl.scheduler.ScheduleAuditLogSupportImpl.addResult(ScheduleAuditLogSupportImpl.java:152)
	at com.enonic.xp.impl.scheduler.ScheduleAuditLogSupportImpl.doUpdate(ScheduleAuditLogSupportImpl.java:87)
	at com.enonic.xp.impl.scheduler.ScheduleAuditLogSupportImpl.lambda$modify$2(ScheduleAuditLogSupportImpl.java:77)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by rymsha** [comment]
Exception is caused by another bug #9049 It does not affect Task execution. Need more info.

**Comment by rymsha** [comment]
RescheduleTask died (maybe long ago):
OSGi is dynamic and Services may be unavailable for a short time
When task tried to access it, Exception was thrown 
This exception must be handled in RescheduleTask, but instead it was rethrown (regression of #8839)
This puts RescheduleTask into limbo state: it is still registered, but does not do anything (This is documented behavior of Hazelcast)

**Comment by rymsha** [comment]
The fix is to restore behavior before #8839 . RescheduleTask  must catch all Throwables and try to log them.
